parser.go

```diff
@@ -1507,103 +1505,145 @@ func (p *parser) getTypeSpec(pkgPath, pkgName, typeName string) (*ast.TypeSpec,
-	if astFields == nil {
-		return
-	}
-	structSchema.Properties = orderedmap.New()
-	if structSchema.DisabledFieldNames == nil {
-		structSchema.DisabledFieldNames = map[string]struct{}{}
+func (p *parser) parseAstFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
+	for _, astField := range astFields {
+		p.parseAstField(pkgPath, pkgName, structSchema, astField)
 	}
+}
 
-	for _, astField := range astFields {
-		fieldSchema := &SchemaObject{}
-		typeAsString := p.getTypeAsString(astField.Type)
-		typeAsString = strings.TrimLeft(typeAsString, "*")
-		isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
-		isInterface := strings.HasPrefix(typeAsString, "interface{}")
-		if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+func (p *parser) parseAstField(pkgPath, pkgName string, structSchema *SchemaObject, astField *ast.Field) {
+	fieldSchema := &SchemaObject{}
+	typeAsString := p.getTypeAsString(astField.Type)
+	isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
+	isInterface := strings.HasPrefix(typeAsString, "interface{}")
+	if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+		splitType := strings.Split(typeAsString, "]")
+		if len(splitType) > 1 {
+			valueType := splitType[1]
+			if !isBasicGoType(valueType) {
+				if _, ok := p.KnownIDSchema[valueType]; ok {
+					nestedType := p.getTypeAsString(splitType[1])
+					setNestedFieldSchemaProps(splitType[0], nestedType, fieldSchema, structSchema)
+				} else {
+					fieldSchemaObjectID, err := p.registerType(pkgPath, pkgName, typeAsString)
+					fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
+					if err != nil {
+						p.debug(err)
+						return
+					}
+					if fieldSchemaObjectID != "" {
+						structSchema.Ref = addSchemaRefLinkPrefix(fieldSchemaObjectID)
+					}
+				}
+			} else {
+				var err error
+				fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
+				if err != nil {
+					p.debug(err)
+					return
+				}
+			}
+		} else {
 			var err error
 			fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
 			if err != nil {
 				p.debug(err)
 				return
 			}
-		} else if !isBasicGoType(typeAsString) {
-			fieldSchemaSchemeaObjectID, err := p.registerType(pkgPath, pkgName, typeAsString)
-			if err != nil {
-				p.debug("parseSchemaPropertiesFromStructFields err:", err)
+		}
+	} else if !isBasicGoType(typeAsString) {
+		fieldSchemaObjectID, err := p.registerType(pkgPath, pkgName, typeAsString)
+		if err != nil {
+			p.debug("parseSchemaPropertiesFromStructFields err:", err)
+		} else {
+			fieldSchema.ID = fieldSchemaObjectID
+			foundSchema, ok := p.KnownIDSchema[fieldSchemaObjectID]
+			if ok {
+				if astField.Tag != nil && !isBasicGoType(foundSchema.Format) && foundSchema.Properties == nil { // inlined structs do not support tagging.
```

Comment:
What's the reason for this change? And when you say "inlined struct" do you mean:

```
type Foo struct {
  Bar struct { Baz string }
}
```

Reply:
This was a left over for `required` I ended up removing even more from that section of code and the spec against test and internally still generates the same changes.